[BUG] - MockBarSeriesBuilder and MockBarBuilder using inconsistent time

**I have checked existing issues and wiki**
- [x] I could not find similar [issues](https://github.com/ta4j/ta4j/issues?utf8=%E2%9C%93&q=)
- [x] I could not find a solution in the [wiki](https://ta4j.github.io/ta4j-wiki/) or [faq section](https://ta4j.github.io/ta4j-wiki/FAQ.html)

**Describe the bug**
There is an inconsistency on the time scale used between MockBarBuilder and MockBarSeriesBuilder. One is using today's date and the other is using beginning of epoch times. 
![Screenshot from 2024-12-06 16-51-21](https://github.com/user-attachments/assets/684d1eca-e681-4b5f-8cbc-05f3c4156bb2)
![Screenshot from 2024-12-06 16-51-46](https://github.com/user-attachments/assets/005f5a48-2272-4902-82b6-b1ae2a2cfde7)

The test RecentSwingHighIndicatorTest uses both. New versions of Maven runs the test in a different order and if you run the test alone the test will fail with this error:
```
[INFO] 
[ERROR] Errors: 
[ERROR]   RecentSwingHighIndicatorTest.testCalculate_OnMovingBarSeries_ReturnsValue:211 » IllegalArgument Cannot add a bar with end time:1975-02-28T00:00:00Z that is <= to series end time: 2024-12-07T00:12:00.897172497Z
[ERROR]   RecentSwingHighIndicatorTest.testCalculate_OnMovingBarSeries_ReturnsValue:211 » IllegalArgument Cannot add a bar with end time:1975-07-31T00:00:00Z that is <= to series end time: 2024-12-07T00:12:00.900613375Z
[INFO] 
[ERROR] Tests run: 1253, Failures: 0, Errors: 2, Skipped: 0
```

The date from MockBarSeriesBuilder is newer than the date from MockBarBuilder

![Screenshot from 2024-12-06 16-47-37](https://github.com/user-attachments/assets/fd9539a9-0314-474f-982e-74ede4dde0c5)

The reason a `mvn clean install` works, is that enough Bars are created so that this is not a problem.
![Screenshot from 2024-12-06 16-55-46](https://github.com/user-attachments/assets/011f4983-a2a8-4ecc-92d6-a10ad452cde2)

So the test is not deterministic. 

**To Reproduce**
Steps to reproduce the behavior:
Run the RecentSwingHighIndicatorTest alone. Test will fail.


**Expected behavior**
A clear and concise description of what you expected to happen.
Tests should pass.

**Additional context**
I think there needs to be consistency in the time scale and the counter between MockBarBuilder and MockBarSeriesBuilder.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -16,6 +16,7 @@ Changelog for `ta4j`, roughly following [keepachangelog.com](http://keepachangel
 - Fixed `SMAIndicatorTest` to set the endTime of the next bar correctly
 - Fixed `SMAIndicatorMovingSeriesTest` to set the endTime of the next bar correctly
 - Use UTC TimeZone for `AroonOscillatorIndicatorTest`, `PivotPointIndicatorTest`
+- Fixed `RecentSwingHighIndicatorTest` to create bars consistently
 
 ### Changed
 - Updated **jfreechart** dependency in **ta4j-examples** project from 1.5.3 to 1.5.5 to resolve [CVE-2023-52070](https://ossindex.sonatype.org/vulnerability/CVE-2023-6481?component-type=maven&component-name=ch.qos.logback%2Flogback-core)

--- a/ta4j-core/src/test/java/org/ta4j/core/indicators/RecentSwingHighIndicatorTest.java
+++ b/ta4j-core/src/test/java/org/ta4j/core/indicators/RecentSwingHighIndicatorTest.java
@@ -33,6 +33,7 @@ import org.junit.Before;
 import org.junit.Test;
 import org.ta4j.core.Bar;
 import org.ta4j.core.BarSeries;
+import org.ta4j.core.BaseBarSeriesBuilder;
 import org.ta4j.core.Indicator;
 import org.ta4j.core.indicators.helpers.ClosePriceIndicator;
 import org.ta4j.core.indicators.helpers.HighPriceIndicator;
@@ -202,8 +203,9 @@ public class RecentSwingHighIndicatorTest extends AbstractIndicatorTest<Indicato
 
     @Test
     public void testCalculate_OnMovingBarSeries_ReturnsValue() {
-        // movingSeries: [1]
-        BarSeries movingSeries = new MockBarSeriesBuilder().withNumFactory(numFactory).withData(1).build();
+        // movingSeries: [1]    	
+        BarSeries movingSeries = new BaseBarSeriesBuilder().withNumFactory(numFactory).build();
+        movingSeries.addBar(new MockBarBuilder(numFactory).closePrice(1).build());
         ClosePriceIndicator closePrice = new ClosePriceIndicator(movingSeries);
         RecentSwingHighIndicator swingHighIndicator = new RecentSwingHighIndicator(closePrice, 1, 1, 0);
 

--- a/ta4j-core/src/test/java/org/ta4j/core/indicators/RecentSwingHighIndicatorTest.java
+++ b/ta4j-core/src/test/java/org/ta4j/core/indicators/RecentSwingHighIndicatorTest.java
@@ -202,7 +202,7 @@ public class RecentSwingHighIndicatorTest extends AbstractIndicatorTest<Indicato
 
     @Test
     public void testCalculate_OnMovingBarSeries_ReturnsValue() {
-        // movingSeries: [1]    	
+        // movingSeries: [1]
         BarSeries movingSeries = new MockBarSeriesBuilder().withNumFactory(numFactory).build();
         movingSeries.addBar(new MockBarBuilder(numFactory).closePrice(1).build());
         ClosePriceIndicator closePrice = new ClosePriceIndicator(movingSeries);

--- a/ta4j-core/src/test/java/org/ta4j/core/indicators/RecentSwingHighIndicatorTest.java
+++ b/ta4j-core/src/test/java/org/ta4j/core/indicators/RecentSwingHighIndicatorTest.java
@@ -204,7 +204,7 @@ public class RecentSwingHighIndicatorTest extends AbstractIndicatorTest<Indicato
     @Test
     public void testCalculate_OnMovingBarSeries_ReturnsValue() {
         // movingSeries: [1]    	
-        BarSeries movingSeries = new BaseBarSeriesBuilder().withNumFactory(numFactory).build();
+        BarSeries movingSeries = new MockBarSeriesBuilder().withNumFactory(numFactory).build();
         movingSeries.addBar(new MockBarBuilder(numFactory).closePrice(1).build());
         ClosePriceIndicator closePrice = new ClosePriceIndicator(movingSeries);
         RecentSwingHighIndicator swingHighIndicator = new RecentSwingHighIndicator(closePrice, 1, 1, 0);

--- a/ta4j-core/src/test/java/org/ta4j/core/indicators/RecentSwingHighIndicatorTest.java
+++ b/ta4j-core/src/test/java/org/ta4j/core/indicators/RecentSwingHighIndicatorTest.java
@@ -33,7 +33,6 @@ import org.junit.Before;
 import org.junit.Test;
 import org.ta4j.core.Bar;
 import org.ta4j.core.BarSeries;
-import org.ta4j.core.BaseBarSeriesBuilder;
 import org.ta4j.core.Indicator;
 import org.ta4j.core.indicators.helpers.ClosePriceIndicator;
 import org.ta4j.core.indicators.helpers.HighPriceIndicator;

